internals/overlord/state/notices_test.go

```diff
@@ -96,6 +98,7 @@ func (s *noticesSuite) TestUnmarshal(c *C) {
 	n := noticeToMap(c, notice)
 	c.Assert(n, DeepEquals, map[string]any{
 		"id":             "1",
+		"user":           1000.0,
```

Comment:
For both `users` and `occurrences`, the notice fields must be unmarshalled from an int, but the marshal code produces a float (due to the nature of json numeric encoding). This means that json which is produced by marshalling a notice cannot be unmarshalled back into a notice.

Since this affects `occurrences` as well, and is very prevalent (all the floats in the json in test code), I propose fixing this in a subsequent PR, assuming it is something we wish to address. I think the two options would be:
1. making `jsonNotice` use `float64` for both `user` and `occurrences`, and convert between `int64` and `float64` in the `Notice.{Marshal,Unmarshal}` code.
2. making `jsonNotice` use `string` for both `user` and `occurrences`, and convert as above -- json may be unhappy with strings which are numbers, however.

Reply:
> the notice fields must be unmarshalled from an int, but the marshal code produces a float 

afaiu it's actually the unmarshaling to any that produces a float here, not the marshaling. So I'm not sure why we would need to change the marshaling side in general?

Reply:
Yes, you're completely right. I had tested this by making the input json to `Unmarshal` use floats, but that's inside a string, hence why it failed for me. I tested locally creating a notice, Marshalling it, then Unmarshalling it, and everything works as expected.